core/src/main/java/net/adamcin/oakpal/core/OakMachine.java

```diff
@@ -360,6 +367,17 @@ public Builder withInstallHookPolicy(final InstallHookPolicy installHookPolicy)
             return this;
         }
 
+        /**
+         * Specify an NodeStore for the scan.
+         *
+         * @param nodeStore the NodeStore
+         * @return my builder self
+         */
+        public Builder withNodeStore(final NodeStore nodeStore) {
```

Comment:
We should probably add a javadoc warning that the provided NodeStore will be mutated, and therefore cannot be guaranteed to be in the same initial state between executions of the same OakMachine. Another thought: Perhaps this should be a Supplier<NodeStore> instead, in case some reinitialization logic needs to be executed?